### PR TITLE
chore: enhance error message clarity and verbosity [ROW-252]

### PR DESCRIPTION
**Pull request by RgnDunes**
# Description

This pull request refactors the error messages throughout the codebase to be more verbose and descriptive. The changes aim to provide clearer and more detailed information about the errors, making it easier for developers to understand what went wrong and how to fix it.

Changes include:
 - Updated all throw new Error statements to include detailed descriptions of the error conditions.
 - Included specific invalid values in the error messages for better debugging.
 - Added guidance in the error messages on what constitutes valid inputs.
 - Ensured consistency in the format and wording of error messages across different modules.

## Changes Made

List the main changes made in this pull request.

| Title                  | --- |
| ---------------------- | --- |
| JIRA link              | https://razorpay.atlassian.net/browse/ROW-252  |
| Slack thread (if any)  | https://razorpay.slack.com/archives/C072GV03Z6D/p1719845063234929?thread_ts=1719492341.561969&cid=C072GV03Z6D  |
| Product spec           | NA  |
| Tech spec/One-pager    | NA  |
| Bundle Size Difference | NA  |
| Fixes Issue            | NA  |

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Is your change backward compatible ?
- [x] Tested on major browsers (Chrome, Firefox, Safari, IE) ?
- [x] Tested in a consumer application(s) ?

## Additional Notes

Any additional information that would be helpful for the reviewer.

## Checklist:

- [x] Add Jira ID(s) in PR title and in the description?
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code is written in TypeScript and follows the coding standards of the project.
- [x] I have added relevant documentation and updated the README if necessary.
- [x] My commit messages are clear and follow the project's commit message conventions.
- [ ] Is any external library added?
- [x] My changes do not introduce any new runtime errors or warnings.
- [x] All relevant unit tests have been added/updated in my PR for the change.
- [ ] Any Screenshots (mobile & desktop) required for PR? If yes, have you added the respective screenshots ?
- [x] Any manual dev testing done by you on beta/func environment?
- [x] Reviewer added (SLA 2 days)

## Reviewer Checklist

- [ ] Sufficient QA/Dev-Testing is done with proof (test cases list)
- [ ] Ensure that the change in bundle size falls within the acceptable range.

---

## PR Title Format

Format: `<type>: <subject>`

Types can be as follows:

- `feat`: (new feature for the user, not a new feature for build script)
- `fix`: (bug fix for the user, not a fix to a build script)
- `docs`: (changes to the documentation)
- `style`: (formatting, missing semi colons, etc; no production code change)
- `refactor`: (refactoring production code, eg. renaming a variable)
- `test`: (adding missing tests, refactoring tests; no production code change)
- `chore`: (updating grunt tasks etc; no production code change)

### CODE REVIEW

File: packages/i18nify-js/src/modules/currency/convertToMinorUnit.ts [hunk 1]
```diff
@@ -22,8 +22,11 @@ const convertToMinorUnit = (
 ): number => {
   const currencyInfo = CURRENCY_INFO[options.currency];
 
-  if (!currencyInfo)
-    throw new Error(`Unsupported currency ${String(options.currency)}`);
+  if (!options.currency || !currencyInfo) {
+    throw new Error(
+      `The provided currency code is either empty or not supported. The received value was ${(options.currency as any) === '' ? 'an empty string' : `: ${String(options.currency)}`}. Please ensure you pass a valid currency code.`,
+    );
```
Comment (tarun-khanna) [inline packages/i18nify-js/src/modules/currency/convertToMinorUnit.ts:28]: Can we attach a link of valid currency codes ?

File: packages/i18nify-js/src/modules/currency/getCurrencySymbol.ts [hunk 2]
```diff
@@ -6,7 +6,10 @@ const getCurrencySymbol = (currencyCode: CurrencyCodeType): string => {
   const currencyInformation = CURRENCY_INFO;
   if (currencyCode in currencyInformation)
     return currencyInformation[currencyCode]?.symbol;
-  else throw new Error(`Invalid currencyCode: ${String(currencyCode)}`);
+  else
+    throw new Error(
+      `The provided currency code is invalid. The received value was: ${String(currencyCode)}. Please ensure you pass a valid currency code.`,
```
Comment (tarun-khanna) [inline packages/i18nify-js/src/modules/currency/getCurrencySymbol.ts:11]: can attach currency code list link

File: packages/i18nify-js/src/modules/geo/getCities.ts [hunk 3]
```diff
@@ -21,7 +21,11 @@ const getCities = (
   const stateCode = _stateCode && _stateCode.toUpperCase();
 
   if (!I18NIFY_DATA_SUPPORTED_COUNTRIES.includes(countryCode)) {
-    return Promise.reject(`Invalid country code: ${countryCode}`);
+    return Promise.reject(
+      new Error(
+        `Invalid country code: ${countryCode}. Please ensure you provide a valid country code.`,
```
Comment (tarun-khanna) [inline packages/i18nify-js/src/modules/geo/getCities.ts:26]: country codes link

File: packages/i18nify-js/src/modules/.internal/utils/getIntlInstanceWithOptions.ts [hunk 4]
```diff
@@ -20,7 +20,10 @@ export const getIntlInstanceWithOptions = (
     intlOptions.currency = (options.currency || intlOptions.currency) as string;
   }
 
-  if (!locale) throw new Error('Pass valid locale !');
+  if (!locale)
+    throw new Error(
+      `The provided locale value is invalid. The received value was: ${locale}. Please ensure you pass a correct locale string for proper formatting.`,
```
[the remark this answers is not in the document]
Comment (RgnDunes) [reply packages/i18nify-js/src/modules/.internal/utils/getIntlInstanceWithOptions.ts:25]: We don't have any such list maintained in our codebase.

**Review by tarun-khanna — APPROVED** [review]

**Pull request merged**